source/simulator/stokes_matrix_free.cc

```diff
@@ -1440,14 +1440,18 @@ namespace aspect
             SymmetricTensor<2,dim,VectorizedArray<double>> sym_grad_u =
                                                           velocity.get_symmetric_gradient (q);
             VectorizedArray<double> pres = pressure.get_value(q);
-            VectorizedArray<double> div = -trace(sym_grad_u);
-            pressure.submit_value   (-1.0*sim.pressure_scaling*div, q);
+            VectorizedArray<double> div = trace(sym_grad_u);
+            pressure.submit_value   (sim.pressure_scaling*div, q);
 
             sym_grad_u *= cell_viscosity_x_2;
 
             for (unsigned int d=0; d<dim; ++d)
               sym_grad_u[d][d] -= sim.pressure_scaling*pres;
 
+            if (sim.material_model->is_compressible())
```

Comment:
I wonder if this will result in a virtual function call in the inner-most loop. Can you create a ``bool`` and move this call outside the loops?